pkg/common/patcher/patcher.go

```diff
@@ -131,6 +140,16 @@ func (p *patcher[R, Sp, St]) RemoveFinalizer(ctx context.Context, object R, fina
 		if errors.IsNotFound(err) {
 			return nil
 		}
+
+		// If it's a conflict, we need to get the latest version of the object and try again.
+		if errors.IsConflict(err) {
```

Comment:
it seems too greedy, you have to set a backoff time. Also instead of jsonMergePatch, should we use jsonPatch here?

Reply:
but setting a backoff time will make the retry slow, which is not good for deleting the klusterlet CR.
can we set a retry times(eg.3 times) into the ctx, and if it exceeds, we stop?